module/VuFind/src/VuFind/Controller/UpgradeController.php

```diff
@@ -938,4 +946,53 @@ protected function fixshortlinks()
             );
         }
     }
+
+    /**
+     * Check for deprecated and insecure use of blowfish encryption
+     *
+     * @return string|null
+     */
+    protected function criticalCheckForBlowfishEncryption()
+    {
+        $config = $this->getConfig();
+
+        if (isset($config->Authentication->ils_encryption_algo)
+            && $config->Authentication->ils_encryption_algo == "blowfish"
+        ) {
+            return 'CriticalFixBlowfish';
+        }
+
+        return null;
+    }
+
+    /**
+     * Lead users through the steps required to replace blowfish quickly and easily
+     *
+     * @return mixed
+     */
+    public function criticalFixBlowfishAction()
+    {
+        // Test that blowfish is still working
+        $blowfishIsWorking = true;
+        try {
+            $newcipher = new BlockCipher(new Openssl(['algorithm' => 'blowfish']));
+            $newcipher->setKey('akeyforatest');
+            $newcipher->encrypt('youfoundtheeasteregg!');
+        } catch (Exception $e) {
+            $blowfishIsWorking = false;
+        }
+
+        // Get key for example command
+        $config = $this->getConfig();
+        $oldkey = $config->Authentication->ils_encryption_key;
```

Comment:
As noted in the template, I don't think we actually need to expose the old key, which is a good thing, because sending the real current key to the web interface under any circumstances is pretty dangerous (in case somebody forgets to disable auto-config).